test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatHelpOutputTests.cs

```diff
@@ -82,5 +101,32 @@ public void MainInternal_OnParsingSuccess_DoesNotShowsHelp(string[] args)
             Assert.DoesNotContain("Usage", output);
             Assert.Equal(1, exitCode);
         }
+
+        [Fact]
+        public void MainInternal_ShowsHelp()
+        {
+            // Arrange
+            var originalConsoleOut = Console.Out;
+            using var consoleOutput = new StringWriter();
+            Console.SetOut(consoleOutput);
+            var log = new TestCommandOutputLogger(_testOutputHelper);
+
+            // Act
+            var exitCode = Program.MainInternal(Array.Empty<string>(), log);
+            Console.SetOut(originalConsoleOut);
+
+            // Assert
+            var output = consoleOutput.ToString();
+
+            var commandPattern = @"^\s{2}(\w+)\s{2,}"; // Matches lines starting with two spaces, a word (command), followed by at least two spaces
+            IEnumerable<string> matches = Regex.Matches(output, commandPattern, RegexOptions.Multiline).Select(m => m.ToString().Trim());
+
+            foreach (var command in HelpCommands)
+            {
+                Assert.Contains(command, matches);
```

Comment:
I'm thinking about what kind of regressions this test will catch, and what it will miss.

It will catch someone removing an existing command from the help. For example, if one command is ported from the existing CLI parser to System.CommandLine and the person removes the comand registration from the `dotnet nuget *` code path, this test will fail, telling the developer not to do that.

This test will not catch the scenario where a brand new command is implemeneted and is special cased like config and why are, so they don't get registered through the `dotnet nuget *` code path. That's exactly the scenario for `config` and `why` commands that this PR is fixing.

Since commands don't need to implement an interface, or extend a base class, I don't have any ideas for how we could improve the test to catch the second scenario. I just wanted to raise awareness, in case someone else reviewing has an idea for an easy way to do it. My philosophy for bug fixes is that any tests added should have prevented the bug in the first place, had the test been in place before the bug was introduced. But this test unfortunately would not have, and I can't think of a nice way to do so.   Maybe this is just a temporary thing until the System.CommandLine migration is complete.

Reply:
I changed the comparison a little bit, I'll check that both have the same strings in the same order and the sizes are the same.

This could me improved later when we finish the migration because there is no nice way to compare it :(